include/sys/sa_impl.h

```diff
@@ -114,6 +114,21 @@ typedef struct sa_idx_tab {
 	uint32_t	*sa_idx_tab;	/* array of offsets */
 } sa_idx_tab_t;
 
+#define	SA_IDX_ENTRIES		5
+
+typedef struct sa_idx_cache {
+	sa_idx_tab_t	*sac_idx[SA_IDX_ENTRIES];
+	sa_lot_t	*sac_lot[SA_IDX_ENTRIES];
+	int		sac_head;
```

Comment:
yes, makes sense. I was going to change SA_IDX_ENTRIES so that the whole structure fit a cacheline..